ui/ui-components/pages/apps.tsx

```diff
@@ -12,20 +13,29 @@ import AppIcon from "../components/AppIcon";
 import StateBadge from "../components/badges/StateBadge";
 import { Models, Actions } from "../utils/apiData";
 import { formatTimestamp } from "../utils/formatTimestamp";
-import { ErrorHandler } from "../utils/errorHandler";
-import { SuccessHandler } from "../utils/successHandler";
 import LinkButton from "../components/LinkButton";
 import LoadingButton from "../components/LoadingButton";
 import { grouparooUiEdition } from "../utils/uiEdition";
 import { formatName } from "../utils/formatName";
+import { GrouparooPage } from "../types/app";
 
-export default function Page(props) {
-  const {
-    errorHandler,
-    successHandler,
-  }: { errorHandler: ErrorHandler; successHandler: SuccessHandler } = props;
+export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
+  const { execApi } = UseApi(ctx);
+  const { limit, offset } = ctx.query;
+  const { apps, total } = await execApi<Actions.AppsList>("get", `/apps`, {
+    limit,
+    offset,
+  });
+  return { props: { apps, total } };
+};
+
+const Page: GrouparooPage<typeof getServerSideProps> = ({
+  errorHandler,
+  successHandler,
+  ...props
+}) => {
   const router = useRouter();
-  const { execApi } = UseApi(props, errorHandler);
+  const { execApi } = UseApi(undefined, errorHandler);
```

Comment:
What if execApi was passed in from _App, already configured to use the errorHandler?  We might want to refactor that first.  Ideally, the new API would provide it to the component already initialized, and then we could override the  default  error handler as needed

Reply:
Now I believe that we should lean in on the hooks pattern instead of passing these down through props. For these examples it's a bit ugly to pass undefined but I believe we can clean up useAPI to be more hook friendly